lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -515,20 +521,24 @@ struct AIEObjectFifoStatefulTransformPass
       }
     }
     createBd(builder, acqLock, acqMode, acqLockAction, relLock, relMode, buff,
-             offset, len, succ, dims);
+             offset, len, succ, dims, padDimensions);
   }
 
   /// Function that either calls createAIETileDMA(), createShimDMA() or
   /// createMemTileDMA() based on op tile row value.
   void createDMA(DeviceOp &device, OpBuilder &builder, ObjectFifoCreateOp op,
                  DMAChannelDir channelDir, int channelIndex, int lockMode,
-                 BDDimLayoutArrayAttr dims) {
+                 BDDimLayoutArrayAttr dims, BDPadLayoutArrayAttr pad_dims) {
     if (op.getProducerTileOp().isShimTile()) {
       createShimDMA(device, builder, op, channelDir, channelIndex, lockMode,
                     dims);
+    } else if (op.getProducerTileOp().isMemTile() &&
+               channelDir == DMAChannelDir::MM2S && pad_dims) {
+      createMemTileDMA(device, builder, op, channelDir, channelIndex, lockMode,
+                       dims, pad_dims);
     } else if (op.getProducerTileOp().isMemTile()) {
       createMemTileDMA(device, builder, op, channelDir, channelIndex, lockMode,
-                       dims);
+                       dims, nullptr);
```

Comment:
I think it might look a little better as:
```
} else if (op.getProducerTileOp().isMemTile()) {
        BDPadLayoutArrayAttr padDims = nullptr;
        if (channelDir == DMAChannelDir::MM2S && pad_dims)
                padDims = pad_dims;
        createMemTileDMA(device, builder, op, channelDir, channelIndex, lockMode,
                dims, padDims);
} 
```